invenio_records_lom/services/components.py

```diff
@@ -26,37 +26,50 @@ class ResourceTypeComponent(ServiceComponent):
 
     new_version_skip_fields = ["publication_date", "version"]
 
-    # pylint: disable-next=unused-argument
     def create(
-        self, identity: Identity, data: dict = None, record: Record = None, **kwargs
+        self,
+        identity: Identity,
+        data: dict = None,
+        record: Record = None,
+        errors=None,  # pylint: disable=unused-argument
+        **kwargs,
     ):
         """Inject parsed resource_type to the record."""
         record.resource_type = data.get("resource_type", {})
 
-    # pylint: disable-next=unused-argument, arguments-differ
     def update_draft(
-        self, identity: Identity, data: dict = None, record: Record = None, **kwargs
+        self,
+        identity: Identity,
+        data: dict = None,
+        record: Record = None,
+        errors=None,
     ):
         """Inject parsed resource_type to the record."""
         record.resource_type = data.get("resource_type", {})
 
-    # pylint: disable-next=unused-argument
     def publish(
-        self, identity: Identity, draft: Record = None, record: Record = None, **kwargs
+        self,
+        identity: Identity,
```

Comment:
this argument is clearly not used

Why doesn't pylint complain here?

Reply:
yes, i know. i also don't understand in detail. my theory is that pylint makes a difference between parameter with and without default values in the inheritance tree of the methods, but i haven't found a proof about this.

Reply:
it's because the parent-class's method with the same name has that argument: [relevant pylint-code](https://github.com/PyCQA/pylint/blob/v3.0.0-a5/pylint/checkers/variables.py#L2377)